system/middleware.go

```diff
@@ -98,6 +150,11 @@ func strInSlice(strs []string, str string) int {
 
 func (application *Application) ApplyCsrfProtection(c *web.C, h http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		// disable CSRF for API requests
+		if c.Env["IsAPI"].(bool) {
```

Comment:
I cool with a nil check here.  This is the spot to give a message about what went wrong (middleware order)